spec/controllers/applications/process_controller_spec.rb

```diff
@@ -144,6 +144,36 @@
     end
   end
 
+  describe 'GET #summary_save' do
+    let(:current_time) { Time.zone.now }
+    let(:user) { create :user }
+    let(:application) { create :application_full_remission }
+
+    before do
+      allow(Application).to receive(:find).with(application.id).and_return(application)
+
+      Timecop.freeze(current_time) do
+        sign_in user
+        get :summary_save, application_id: application.id
+      end
+    end
+
+    it 'returns the correct status code' do
+      expect(response).to have_http_status(302)
+    end
+
+    it 'redirects to the confirmation page' do
+      expect(response).to redirect_to(application_confirmation_path(application.id))
+    end
+
+    it 'updates the payment completed_at and completed_by' do
+      application.reload
```

Comment:
Same here, if you changed the `application` to be build as stubbed, then you don't have to reload it here.